airflow/providers/apache/druid/provider.yaml

```diff
@@ -51,6 +51,10 @@ hooks:
 hook-class-names:
   - airflow.providers.apache.druid.hooks.druid.DruidDbApiHook
 
+connection-types:
```

Comment:
Can we not remove the top-level `hook-class-names` now, the deprecation will allow custom providers, but for current ones we can remove, right? 

Probably not -- to keep double-compatibility between Airflow version and Provider too

Reply:
Not really, the providers must work with older Airlfows too :(

Reply:
Yea figured :(

Reply:
We will be able to do it when we add dependency on Airflow 2.2 but this will happen only after some time when we will have good reason for it. And even then the code to parse it we will be able to remove only in Airflow 3 really.

I treat it also partially as exercise to learn and solve potential deprecation/feature patterns we might have for providers vs. Airflow. This is a rather  straightforward change but we might learn how to deal with evolving providers smoothly. 

We do not "have to" do it now (it's not a huge optimisation), but I figured this might be good idea to implement it and see what it brings. I think that this is a good outcome from that as well.